src/main/connector/system/WindowsProcess/WindowsProcess.yaml

```diff
@@ -100,7 +100,23 @@ monitors:
             FROM
               processQuery
             WHERE
-              Name IN ${var::processName}
+              Name IN ${var::processNames}
+
+            # The value of the variable 'processNames'
+            # is set in MetricsHub configuration file 'metricshub.yaml'. Here is an example:
+            # resources:
+            #  localhost:
+            #    attributes:
+            #      host.name: localhost:
+            #      host.type: windows
+            #    protocols:
+            #      wmi:
+            #        timeout: 120
+            #    connectors: ["+WindowsProcess"]
+            #    variables:
+            #      WindowsProcess:
+            #        processNames: ('msedge.exe', 'process2.exe')

```

Comment:
Really, this is an open door to SQL injection through the `processNames` variable.

Let's be consistent on the various process monitoring connectors we have and allow filtering through dedicated variables for the different fields:

* `processNameRegex`
* `processCommandRegex`
* `processUsernameRegex`

We could even drop the `process` prefix, as the variable applies to this connector only:

* `nameMatches`
* `commandMatches`
* `usernameMatches`

And use `keepOnlyMatchingLines` compute steps.

Other question: is it possible to define default values for variables in a connector?
Ideally, the connector would be able to declare its variables like below:

```yaml
variables:
  nameMatches:
    description: Regular expression that process names must match to be monitored
    defaultValue: .*
  commandMatches:
    description: Regular expression that the command line of the process must match to be monitored
    defaultValue: .*
etc.
```

This would help with the documentation of the connector as well!

Reply:
@bertysentry 

> Other question: is it possible to define default values for variables in a connector?

Yes, it's possible but not yet implemented in MetricsHub

Reply:
@MedMaalej Can you create the issue for MetricsHub (if it doesn't exist already)? We need this feature to create proper connectors that rely on variables. Thank you!

Reply:
@bertysentry Sorry, I have just seen your comment. Here is the corresponding Github issue that I have just created https://github.com/sentrysoftware/metricshub/issues/374